FlashLFQ/Bayesian Protein Quantification/ProteinQuantificationEngine.cs

```diff
@@ -0,0 +1,608 @@
+
+            Parallel.ForEach(Partitioner.Create(0, proteinList.Count), new ParallelOptions { MaxDegreeOfParallelism = MaxThreads }, partitionRange =>
+            {
+                for (int i = partitionRange.Item1; i < partitionRange.Item2; i++)
+                {
+                    ProteinGroup protein = proteinList[i].Key;
+
+                    List<double> peptideFoldChanges = new List<double>();
+
+                    if (biorepCount > 1)
+                    {
+                        if (!PairedSamples)
+                        {
+                            // get the fold-change measurements for the peptides assigned to this protein between bioreps of this condition
+                            for (int b = 0; b < biorepCount - 1; b++)
+                            {
+                                peptideFoldChanges.AddRange(GetPeptideFoldChangeMeasurements(protein, treatmentCondition, treatmentCondition, b, b + 1));
+                            }
+                        }
+                        else if (PairedSamples)
+                        {
+                            peptideFoldChanges.AddRange(GetPeptideFoldChangeMeasurements(protein, BaseCondition, treatmentCondition, null, null));
+                        }
+                    }
+                    else
+                    {
+                        peptideFoldChanges.AddRange(GetPeptideFoldChangeMeasurements(protein, BaseCondition, treatmentCondition, null, null));
+                    }
+
+                    ProteinQuantificationEngineResult result;
+
+                    if (peptideFoldChanges.Count > 1)
+                    {
+                        // run the Bayesian analysis
+                        result = RunBayesianProteinQuant(peptideFoldChanges, protein, treatmentCondition,
+                            randomSeedsForEachProtein[protein][0], null, false, 500, 500, out var mus);
+
+                        lock (experimentalNullResults)
+                        {
+                            experimentalNullResults.Add(protein.ProteinGroupName, new List<ProteinQuantificationEngineResult> { result });
+                        }
+                    }
+                    else
+                    {
+                        continue;
+                    }
+                }
+            });
+
+            double stdDev = 0;
+
+            // this is a weird case where there are very few proteins in the data...
+            if (experimentalNullResults.Count < 10)
+            {
+                if (experimentalNullResults.Any())
+                {
+                    stdDev = experimentalNullResults.SelectMany(p => p.Value).Select(p => p.StandardDeviationPointEstimate).Average();
+                }
+                else
+                {
+                    // TODO: not sure what to do here.. there were no protein quant results for this condition
```

Comment:
...did you ever get to this? Can you remove the TODO and explain what your solution was, or elaborate on the TODO and  what your work around currently is?

Reply:
I changed the wording of this comment a bit